spec/ysh-builtin-error.test.sh

```diff
@@ -114,7 +114,7 @@ error (100, status=42)
 
 #### Errors cannot take command args
 error uh-oh ('error', status=1)
-## status: 3
+## status: 2
```

Comment:
Hmm why did this one change from 3 to 2?

Reply:
The extra `uh-oh` is registered as a usage error. We raise `error.Usage` on extra word arguments being passed to the `typed_args.Reader`.

This behavior now better matches other builtins (or system tools):

```sh
$ # for reference, on `master`
$ bin/ysh -c 'try { error --not-a-flag ("My error") }; echo $_status'
3
$ bin/ysh -c 'try { source --not-a-flag }; echo $_status'
  try { source --not-a-flag }; echo $_status
               ^~~~~~~~~~~~
[ -c flag ]:1: 'source' got invalid flag '--not-a-flag'
2
```